### PR TITLE
[10.0][IMP] website_portal_medical_prescription_order_line: Add 'Expired' and 'Out of Refills' labels

### PR DESCRIPTION
**Pull request by kellylougheed**
- Adds 'Expired' and 'Out of Refill labels when applicable to prescription order lines on "My Medical"
screen shot 2017-03-22 at 11 59 16 am
- Adds a computed is_expired field to medical_prescription_order_line

### CODE REVIEW

File: website_portal_medical_prescription_order_line/models/medical_prescription_order_line.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    is_expired = fields.Boolean(
+        string='Expired?',
+        compute='_compute_is_expired',
+    )
+
+    @api.one
+    def _compute_is_expired(self):
+        stop = self.date_stop_treatment
+        if stop and type(stop) is str:
+            stop = datetime.datetime.strptime(
```
Comment (lasley) [inline website_portal_medical_prescription_order_line/models/medical_prescription_order_line.py]: `fields.Datetime.from_string`

File: website_portal_medical_prescription_order_line/models/medical_prescription_order_line.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+class MedicalPrescriptionOrderLine(models.Model):
+    _inherit = 'medical.prescription.order.line'
+
+    is_expired = fields.Boolean(
+        string='Expired?',
+        compute='_compute_is_expired',
+    )
+
+    @api.one
```
Comment (lasley) [inline website_portal_medical_prescription_order_line/models/medical_prescription_order_line.py]: `api.one` is deprecated. Use `api.multi` & record loop

File: website_portal_medical_prescription_order_line/models/medical_prescription_order_line.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    def _compute_is_expired(self):
+        stop = self.date_stop_treatment
+        if stop and type(stop) is str:
+            stop = datetime.datetime.strptime(
+                stop, "%Y-%m-%d %H:%M:%S"
+            )
+        for record in self:
```
Comment (lasley) [inline website_portal_medical_prescription_order_line/models/medical_prescription_order_line.py]: You already have the record loop. This wasn't needed with api.one, which loops for you. But it's deprecated.

File: website_portal_medical_prescription_order_line/models/medical_prescription_order_line.py [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+            if stop and stop < datetime.datetime.now():
+                record.is_expired = True
```
Comment (lasley) [inline website_portal_medical_prescription_order_line/models/medical_prescription_order_line.py]: `record.is_expired = stop and stop < now`

File: website_portal_medical_prescription_order_line/models/medical_prescription_order_line.py [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+    @api.one
+    def _compute_is_expired(self):
+        stop = self.date_stop_treatment
+        if stop and type(stop) is str:
+            stop = datetime.datetime.strptime(
+                stop, "%Y-%m-%d %H:%M:%S"
+            )
+        for record in self:
+            if stop and stop < datetime.datetime.now():
```
Comment (lasley) [inline website_portal_medical_prescription_order_line/models/medical_prescription_order_line.py]: `.now()` should be computed before the record loop & assigned to variable for reuse

**Review by lasley — CHANGES_REQUESTED** [review]

**Comment by kellylougheed** [comment]
Changes made @lasley

File: website_portal_medical_prescription_order_line/models/medical_prescription_order_line.py [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+    is_expired = fields.Boolean(
+        string='Expired?',
+        compute='_compute_is_expired',
+    )
+
+    @api.multi
+    def _compute_is_expired(self):
+        for record in self:
+            now = datetime.datetime.now()
+            stop = fields.Datetime.from_string(self.date_stop_treatment)
```
Comment (lasley) [inline website_portal_medical_prescription_order_line/models/medical_prescription_order_line.py]: s/self/record

File: website_portal_medical_prescription_order_line/models/medical_prescription_order_line.py [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+
+    is_expired = fields.Boolean(
+        string='Expired?',
+        compute='_compute_is_expired',
+    )
+
+    @api.multi
+    def _compute_is_expired(self):
+        for record in self:
+            now = datetime.datetime.now()
```
Comment (lasley) [inline website_portal_medical_prescription_order_line/models/medical_prescription_order_line.py]: Move this outside the record loop

**Review by lasley — CHANGES_REQUESTED** [review]

File: website_portal_medical_prescription_order_line/models/medical_prescription_order_line.py [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+
+from odoo import api, fields, models
+
+import datetime
+
+
+class MedicalPrescriptionOrderLine(models.Model):
+    _inherit = 'medical.prescription.order.line'
+
+    is_expired = fields.Boolean(
```
Comment (lasley) [inline website_portal_medical_prescription_order_line/models/medical_prescription_order_line.py]: This should be moved into medical_prescription

**Comment by kellylougheed** [comment]
Changes made @lasley

**Review by lasley — APPROVED** [review]

**Pull request merged**